MRFIXIT: rockchip adds the feature of defining a custom resolution in batocera.conf for retroarch. This messes up the bezel if the defined resolution is different than the normal framebuffer resolution. Pull in this value and subtract it's difference to correct this distortion.
--- a/configgen/generators/libretro/libretroConfig.py
+++ b/configgen/generators/libretro/libretroConfig.py
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 import sys
 import os
+import subprocess
 import batoceraFiles
 import libretroOptions
 from Emulator import Emulator
@@ -361,6 +362,15 @@
     else:
         infos = {}
 
+    ra_width = 0
+    ra_height = 0
+    ra_res = subprocess.Popen(['/usr/bin/batocera-settings', '-command', 'load', '-key','retroarch.resolution'], stdout=subprocess.PIPE).communicate()[0].split("x")
+    if unicode(ra_res[0]).isnumeric():
+      ra_width = infos["width"] - int(ra_res[0].replace("\n",""))
+      if len(ra_res) > 1:
+        if unicode(ra_res[1].replace("\n","")).isnumeric():
+          ra_height = infos["height"] - int(ra_res[1].replace("\n",""))
+
     # if image is not at the correct size, find the correct size
     bezelNeedAdaptation = False
     viewPortUsed = True
@@ -418,8 +423,8 @@
         if bezel_stretch:
             retroarchConfig['custom_viewport_x']      = infos["left"] * wratio
             retroarchConfig['custom_viewport_y']      = infos["top"] * hratio
-            retroarchConfig['custom_viewport_width']  = (infos["width"]  - infos["left"] - infos["right"])  * wratio
-            retroarchConfig['custom_viewport_height'] = (infos["height"] - infos["top"]  - infos["bottom"]) * hratio
+            retroarchConfig['custom_viewport_width']  = (infos["width"] - ra_width  - infos["left"] - infos["right"])  * wratio
+            retroarchConfig['custom_viewport_height'] = (infos["height"] - ra_height - infos["top"]  - infos["bottom"]) * hratio
             retroarchConfig['video_message_pos_x']    = infos["messagex"] * wratio
             retroarchConfig['video_message_pos_y']    = infos["messagey"] * hratio
         else:
@@ -442,8 +447,8 @@
                 yoffset = gameResolution["height"] - infos["height"]
                 retroarchConfig['custom_viewport_x']      = infos["left"] + xoffset/2
                 retroarchConfig['custom_viewport_y']      = infos["top"] + yoffset/2
-                retroarchConfig['custom_viewport_width']  = infos["width"]  - infos["left"] - infos["right"]
-                retroarchConfig['custom_viewport_height'] = infos["height"] - infos["top"]  - infos["bottom"]
+                retroarchConfig['custom_viewport_width']  = infos["width"] - ra_width - infos["left"] - infos["right"]
+                retroarchConfig['custom_viewport_height'] = infos["height"] - ra_height - infos["top"]  - infos["bottom"]
                 retroarchConfig['video_message_pos_x']    = infos["messagex"] + xoffset/2
                 retroarchConfig['video_message_pos_y']    = infos["messagey"] + yoffset/2
 
@@ -474,8 +479,8 @@
         if viewPortUsed:
             retroarchConfig['custom_viewport_x']      = infos["left"]
             retroarchConfig['custom_viewport_y']      = infos["top"]
-            retroarchConfig['custom_viewport_width']  = infos["width"]  - infos["left"] - infos["right"]
-            retroarchConfig['custom_viewport_height'] = infos["height"] - infos["top"]  - infos["bottom"]
+            retroarchConfig['custom_viewport_width']  = infos["width"] - ra_width  - infos["left"] - infos["right"]
+            retroarchConfig['custom_viewport_height'] = infos["height"] - ra_height - infos["top"]  - infos["bottom"]
         retroarchConfig['video_message_pos_x']    = infos["messagex"]
         retroarchConfig['video_message_pos_y']    = infos["messagey"]
 
